sci-rs/src/signal/filter/savgol_filter.rs

```diff
@@ -69,7 +82,12 @@ where
 /// This function is sensitive to f64 and f32 primitives due to use of least squares.
 /// The coefficients may go to zero for higher order polynomials and larger window lengths.
 ///
-pub fn savgol_coeffs_dyn<F>(window_length: usize, polyorder: usize) -> Vec<F>
+pub fn savgol_coeffs_dyn<F>(
+    window_length: usize,
+    polyorder: usize,
+    deriv: usize,
+    delta: F,
+) -> Vec<F>
```

Comment:
Here and for savgol_filter_dyn, we should accept parameters that make it clear what the defaults are like the kwargs. For example, deriv would be an Option<usize> that you would unwrap_or(0) and delta would be an Option<F> that you would unwrap or F::one()